openlibrary/plugins/openlibrary/home.py

```diff
@@ -140,7 +140,7 @@ def loans_carousel(loans=None, css_id="CarouselLoans"):
     """Generates 'Your Loans' carousel on home page"""
     _loans = loans or []
     books = [format_book_data(web.ctx.site.get(loan['book']))
```

Comment:
We may want to check that within the list comprehension web.ctx.site.get(loan['book']) isn't being run 2 times for each book (i.e. in the `if` check and within `format_book_data`). I don't recall if python is smart enough to optimize / cache the check within the list comprehension.

The alternative might be a for loop with a tmp variable for web.ctx.site.get(loan['book']).

    books = []
    for book in books:
        loan_book = web.ctx.site.get(loan['book'])
        if loan_book:
            books.append(format_book_data(loan_book))

Especially since home.py has performance challenges, this may be worth a look.

The plus side is, there can only be a max of n=5 books lent out, which means a max of 10 calls (but calls to web.ctx.site.get do seem to come with non-trivial overhead)